src/notification-banner/notification-banner.tsx

```diff
@@ -40,36 +47,94 @@ export const NBComponent = ({
     // =============================================================================
-    const handleDismiss = () => {
+    const handleDismiss = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
         setVisible(false);
 
         if (dismissible && onDismiss) onDismiss();
     };
 
+    const handleBannerClick = (event: React.MouseEvent) => {
+        if (!onClick) return;
+        event.stopPropagation();
+        onClick();
+    };
+
+    const handleActionButtonOnClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
+        if (!actionButton.onClick) {
+            handleBannerClick(event);
+            return;
+        }
+        actionButton.onClick(event);
+    };
+
     // =============================================================================
     // RENDER FUNCTIONS
     // =============================================================================
     if (!isVisible) return null;
 
+    const renderDismissButton = () => (
+        <StyledIconButton
+            onClick={handleDismiss}
+            id={formatId("dismiss-button", id)}
+            data-testid={formatId("dismiss-button", testId)}
+            focusHighlight={false}
+        >
+            <StyledIcon />
+        </StyledIconButton>
+    );
+
+    const renderActionButton = () => (
+        <ActionButton
+            id={formatId("action-button", id)}
+            data-testid={formatId("action-button", testId)}
+            onClick={handleActionButtonOnClick}
+            {...actionButton}
+        >
+            {actionButton.children}
+        </ActionButton>
+    );
+
+    const renderChildren = () => {
+        if (
+            !maxCollapsedHeight ||
+            !contentRef.current ||
+            (contentHeight && contentHeight < maxCollapsedHeight)
+        ) {
+            return children;
+        } else {
+            return (
+                <div ref={contentRef}>
+                    {children}
+                </div>
+            );
+        }
+    };
+
+    const renderAccessibleBannerButton = () => (<AccessibleBannerButton onClick={handleBannerClick} aria-label={"Clickable banner"}/>);
+
     return (
-        <Wrapper ref={forwardedRef} $sticky={sticky} {...otherProps}>
+        <Wrapper
+            ref={forwardedRef}
+            $sticky={sticky}
+            $clickable={!!onClick}
+            onClick={handleBannerClick}
+            {...otherProps}
+        >
             <Container id={formatId("container", id)}>
                 <TextContainer>
-                    <Content data-testid={formatId("text-content", testId)}>
-                        {children}
+                    <Content
+                        data-testid={formatId("text-content", testId)}
+                        $maxCollapsedHeight={maxCollapsedHeight} 
+                    >
+                        {renderChildren()}
+                        {actionButton && renderActionButton()}
```

Comment:
the action button should be outside of `Content` as well, else it'll get collapsed too

you could move it to be a direct descendent of TextContainer, which then requires a `flex-direction: column`